channeldb/channel.go

```diff
@@ -2470,6 +2486,12 @@ func fetchChanCommitment(chanBucket *bbolt.Bucket, local bool) (ChannelCommitmen
 func fetchChanCommitments(chanBucket *bbolt.Bucket, channel *OpenChannel) error {
 	var err error
 
+	// If this is a restored channel, then we don't have any commitments to
+	// read.
+	if channel.hasChanStatus(ChanStatusRestored) {
```

Comment:
this also could return an error, and lift the conditional above? that way other places that call `fetchChanCommitments` don't succeed silently when the channels have been restored

Reply:
See comment above, the idea is that it does, since we don't write any commitment data for restored channels. Threading through this check allowed me to minimize the number of special cases when reading out channels that are only actually shells.